internal/pkg/earpointer/analysis.go

```diff
@@ -387,6 +428,33 @@ func (vis *visitor) visitExtract(extract *ssa.Extract) {
 		if base.CommaOk && extract.Index == 0 {
 			vis.processIndex(extract, base.X, base.Index)
 		}
+	case *ssa.UnOp:
+		// Handle the channel read case with comma.
+		// Other UnOp cases are handled within "visitUnOp()".
+		// t1 = <-t0,ok
+		// t2 = extract t1 0
+		if base.Op == token.ARROW && base.CommaOk && extract.Index == 0 { // channel read
+			// Use array read to simulate channel read.
+			vis.processHeapAccess(extract, base.X, anyIndexField)
+		}
+	case *ssa.Select:
+		// t1 = select blocking [c1<-t0, <-c2]  // (index int, recvOk bool, r_0 T_0, ... r_n-1 T_n-1)
+		// t2 = extract t1 #n
+		if extract.Index > 1 {
```

Comment:
There is a fair bit of nesting going on here. Consider breaking if the condition isn't met.